template.go

```diff
@@ -4,36 +4,97 @@ import (
 	"bytes"
 	"encoding/json"
 	"errors"
+	"fmt"
 )
 
 // Template represents a Template for launching instances from
 type Template struct {
 	ID               string `json:"id"`
 	Code             string `json:"code"`
 	Name             string `json:"name"`
-	AccountID        string `json:"account_id"`
-	ImageID          string `json:"image_id"`
+	AccountID        string `json:"account_id,omitempty"`
+	ImageID          string `json:"image_id,omitempty"`
 	VolumeID         string `json:"volume_id"`
 	ShortDescription string `json:"short_description"`
 	Description      string `json:"description"`
 	DefaultUsername  string `json:"default_username"`
 	CloudConfig      string `json:"cloud_config"`
 }
 
-// GetTemplateByCode finds the Template for an account with the specified code
-func (c *Client) GetTemplateByCode(code string) (*Template, error) {
+// ListTemplates return all template in system
+func (c *Client) ListTemplates() ([]Template, error) {
 	resp, err := c.SendGetRequest("/v2/templates")
 	if err != nil {
 		return nil, err
 	}
 
 	templates := make([]Template, 0)
-	err = json.NewDecoder(bytes.NewReader(resp)).Decode(&templates)
-	for _, template := range templates {
+	if err := json.NewDecoder(bytes.NewReader(resp)).Decode(&templates); err != nil {
+		return nil, err
+	}
+
+	return templates, nil
+}
+
+// NewTemplate this function will create a new template for the current user
```

Comment:
`// NewTemplate will create a new template for the current user`

No need for a "this function"